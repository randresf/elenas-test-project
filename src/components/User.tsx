import { Box, FormLabel } from '@chakra-ui/core';
import React from 'react';

const User = (data: Record<string, string>) => {
  return <Box>
    <FormLabel>Name: {data.firstName} {data.lastName}</FormLabel>
    <FormLabel>Direccion: {data.address}</FormLabel>
    <FormLabel>Ciudad: {data.city}</FormLabel>
  </Box>
}
export default User