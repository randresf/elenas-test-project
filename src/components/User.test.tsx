import { cleanup, screen, render } from '@testing-library/react';
import React from 'react';
import renderer from 'react-test-renderer'
import User from './User';

afterEach(cleanup)

const defaultData = {
  firstName: 'k',
  lastName: 's',
  address: 'calle 66',
  city: 'medellin'
}

const UserCompo = <User {...defaultData} />

test('matches snapshop', () => {
  const comp = renderer.create(UserCompo);
  let tree = comp.toJSON()
  expect(tree).toMatchSnapshot()

});

test('renders correct info', () => {
  render(UserCompo)
  expect(screen.getByText(defaultData.city, { exact: false })).toBeInTheDocument()
  expect(screen.getByText(defaultData.address, { exact: false })).toBeInTheDocument()
})
