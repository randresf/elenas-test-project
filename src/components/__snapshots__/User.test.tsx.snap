// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshop 1`] = `
<div
  className="css-0"
>
  <label
    className="css-a98wfm"
  >
    Name: 
    k
     
    s
  </label>
  <label
    className="css-a98wfm"
  >
    Direccion: 
    calle 66
  </label>
  <label
    className="css-a98wfm"
  >
    Ciudad: 
    medellin
  </label>
</div>
`;
